### ISSUE TITLE
Trace parser blind to `tool_result.content` — CC 2.1.228 RELOCATED the PostToolUse injection, it still delivers

### ISSUE THREAD

**Issue by zernie** [zernie/agent-lint#231]
> 🔴 **RETRACTED AND REPLACED 2026-09-10.** The original title and body of this issue claimed Claude Code stopped delivering `PostToolUse` `additionalContext` at 2.1.228. **That was false.** The bisect was real; its interpretation was not. What follows is the corrected finding. The original text remains in this issue's edit history.

## What is actually true

Claude Code **2.1.228 relocated** the `PostToolUse` hook payload: instead of arriving as a `text` block, `additionalContext` now arrives inside a **`tool_result` block's `content`**. It is delivered on both 2.1.227 and 2.1.267.

Our measuring instrument could not see it. `flattenContent` in `src/mock-model.ts` read only `.text` off each content block:

```ts
return content
  .map((b) => {
    if (typeof b === "string") return b;
    const t = (b as { text?: unknown }).text;
    return typeof t === "string" ? t : "";   // tool_result carries `content`, not `text` → ""
  })
  .join("");
```

`ContentBlockParam` has 16 variants and **eight of them carry `content` rather than `text`**. All eight flattened to `""`, so `requestContains` — and therefore every probe built on it — was blind to that whole family.

The relocation changed nothing about the block's TYPE, which is why nothing in the type system noticed either.

## Independent verification of the corrected finding

On claude **2.1.267** (the machine's actual binary, not the pin):

| | result |
| --- | --- |
| parser fixed to read `tool_result.content` | probe **green** |
| parser reverted to `.text`-only | probe **red** |
| raw request capture, read by hand | payload **present** in `tool_result.content` |

The version is not the variable. The parser is.

## What was wrong, precisely

Three claims in the original body are false and must not be cited:

1. **The title** — "PostToolUse hook output no longer reaches the model". It does.
2. **The bisect conclusion** — 227-good / 228-bad measured *the probe going blind*, not the product breaking. Both versions deliver.
3. **The "all three channels dead" table** — `exit 2 + stderr` and `decision:block` were measured through the same blind parser, so those rows carry no information either. They are unmeasured, not dead.

Consequently: `claudeCodeHookProtocol.injectableEvents` is **not** false about `PostToolUse`, and no upstream bug report is warranted. One was drafted on the false premise and is cancelled.

## What still stands

- `#224` is untouched by this. A compiled react's `notice()` going to the debug log at exit 0 is a separate, real defect of ours.
- The scope limit stands: every measurement here drives `claude -p` headless. Interactive sessions remain unmeasured.

## Fix

#233 replaces the `.text` probe with an exhaustive `switch` over the real `ContentBlockParam` union (`@anthropic-ai/sdk`, MIT, **type-only** devDependency — erased at compile time), with `const unhandled: never = b` in the `default` so a new SDK variant becomes a `tsc` error rather than a silent `""`.

⚠️ Stated plainly there and repeated here, because it is the lesson rather than the patch: **the type does not catch a field MOVING between existing variants**, which is exactly what 2.1.228 did. That is why the `default` branch serialises an unmodelled block instead of returning `""` — a needle in a shape we do not model still shows up, loudly.

## The class, for the record

This repo already names `fired ≠ landed`. This incident adds a third level: **`landed ≠ seen-by-the-probe`**. The instrument had an unstated assumption about the shape of what it measured, and when the shape moved, the instrument reported a product regression that had not happened.
